source/use-cases/sensitive-data-encryption.txt

```diff
@@ -0,0 +1,312 @@
+
+     - id: java-jsonschema-generation
+       name: "Java"
+       content: |
+
+         The following code creates a collection using the schema above and inserts it into a database using MongoDB.
+         For the "Define a JSON Schema" section we've abstracted the createSecureClient, to focus on creation of a schema.
+         But that function uses getJSONSchema and passes it a dataKey id in order to create a client which uses the JSONSchema
+         we want to use. 
+
+         .. code-block:: java
+
+            import com.mongodb.*;
+            import com.mongodb.client.MongoClient;
+            import com.mongodb.client.MongoClients;
+            import com.mongodb.client.MongoCollection;
+            import com.mongodb.client.model.vault.DataKeyOptions;
+            import com.mongodb.client.vault.ClientEncryptions;
+
+            import org.bson.BsonDocument;
+
+            import java.util.Base64;
+            import java.util.Map;
+
+            import org.json.*;
+
+            public class FieldLevelEncryptionDemo {
+                static String dbName = "fieldLevelEncryptionDemo001";
+                static String collectionName = "patients";
+                public static void main(String[] args) {
+                    MongoCollection secureCollection = createSecureClient().getDatabase(dbName).getCollection(collectionName);
+                    ...
+                }
+                public static JSONObject getEncryptedField(String keyId, String chosenBsonType, Boolean isDeterministic){
+                    JSONObject binaryForKey = new JSONObject();
+                    binaryForKey.put("base64", keyId);
+                    binaryForKey.put("subType", "04");
+                    JSONObject binWrapper = new JSONObject();
+                    binWrapper.put("$binary", binaryForKey);
+                    
+                    JSONObject encryptedFieldWrap = new JSONObject();
+                    JSONObject encrypt = new JSONObject();
+
+                    encrypt.put("bsonType", chosenBsonType);
+                    if(isDeterministic){
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic");
+                    }else{
+                        encrypt.put("algorithm", "AEAD_AES_256_CBC_HMAC_SHA_512-Random");
+                    }
+                    encrypt.put("keyId", new JSONArray().put(binWrapper));
+                    encryptedFieldWrap.put("encrypt", encrypt);
+                    return encryptedFieldWrap;
+                }
+                public static JSONObject getStandardField(String chosenBsonType){
+                    JSONObject standardField = new JSONObject();
+                    standardField.put("bsonType", chosenBsonType);
+                    standardField.put("description", "must be a " + chosenBsonType + " and is required");
+                    return standardField;
+                }
+                public static String getJSONSchema(String keyId){
```

Comment:
Similar comment on the method name prefix `get`; this could be `createJSONSchema`.

Reply:
Fixed